File: app/src/main/java/woo/sopt22/meowbox/View/MeowBoxDetail/DetailViewAdapter.java
package woo.sopt22.meowbox.View.MeowBoxDetail;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.List;

import woo.sopt22.meowbox.R;

public class DetailViewAdapter extends PagerAdapter {
    private LayoutInflater inflater;
    private List<DetailModel> mPageList;
    private Context context;
    private View.OnClickListener onItemClick;

    public DetailViewAdapter(LayoutInflater inflater, List<DetailModel> mPageList) {
        this.inflater = inflater;
        this.mPageList = mPageList;
    }

    @Override
    public int getCount() {
        return mPageList.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        View view = inflater.inflate(R.layout.detail_card_view, container, false);


        ImageView detailCardViewImg = (ImageView) view.findViewById(R.id.detail_card_view_img);
        TextView detailCardViewTxt = (TextView) view.findViewById(R.id.detail_card_view_txt);
        DetailModel item = mPageList.get(position);
        Glide.with(container.getContext()).load(item.getImgUrl()).into(detailCardViewImg);
        detailCardViewTxt.setText(item.getText());

        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((View) object);
    }


    public void setOnItemClickListener(View.OnClickListener l){
        onItemClick = l;
    }
}

File: app/src/main/java/woo/sopt22/meowbox/View/MeowBoxDetail/DetailViewAdapter.kt
package woo.sopt22.meowbox.View.MeowBoxDetail

import android.content.Context
import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.detail_card_view.view.*

import woo.sopt22.meowbox.R

class DetailViewAdapter(private val inflater: LayoutInflater, private val mPageList: List<DetailModel>) : PagerAdapter() {
    private val context: Context? = null
    private var onItemClick: View.OnClickListener? = null

    override fun getCount(): Int {
        return mPageList.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = inflater.inflate(R.layout.detail_card_view, container, false)


        //val detailCardViewImg = view.findViewById<View>(R.id.detail_card_view_img) as ImageView
        //val detailCardViewTxt = view.findViewById<View>(R.id.detail_card_view_txt) as TextView
        val item = mPageList[position]
        Glide.with(container.context).load(item.imgUrl).into(view.detail_card_view_img)
        view.detail_card_view_txt.text = item.text

        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }


    fun setOnItemClickListener(l: View.OnClickListener) {
        onItemClick = l
    }
}